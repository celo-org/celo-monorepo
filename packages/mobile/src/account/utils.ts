import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
