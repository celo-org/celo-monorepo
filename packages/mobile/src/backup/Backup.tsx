import * as React from 'react'
import { connect } from 'react-redux'
import { setBackupCompleted, setBackupDelayed } from 'src/account/actions'
import componentWithAnalytics from 'src/analytics/wrapper'
import { enterBackupFlow, exitBackupFlow } from 'src/app/actions'
import BackupComplete from 'src/backup/BackupComplete'
import BackupIntroduction from 'src/backup/BackupIntroduction'
import BackupPhrase from 'src/backup/BackupPhrase'
import BackupQuestion from 'src/backup/BackupQuestion'
import { createQuizWordList, getStoredMnemonic, selectQuizWordOptions } from 'src/backup/utils'
import { navigateBack } from 'src/navigator/NavigationService'
import { RootState } from 'src/redux/reducers'
import { isBackupTooLate } from 'src/redux/selectors'
import Logger from 'src/utils/Logger'

export const DAYS_TO_BACKUP = 1
export const DAYS_TO_DELAY = 1 / 24 // 1 hour delay

const OPTIONS_PER_QUESTION = 4
const INDICES_TO_TEST = [0, 2, 3, 6]
const NUMBER_OF_TEST_QUESTIONS = INDICES_TO_TEST.length

interface State {
  mnemonic: string
  currentQuestion: number | null
  wordsForBackupQuiz: string[]
}

interface StateProps {
  language: string | null
  backupCompleted: boolean
  backupTooLate: boolean
  backupDelayedTime: number
}

interface DispatchProps {
  setBackupCompleted: typeof setBackupCompleted
  setBackupDelayed: typeof setBackupDelayed
  enterBackupFlow: typeof enterBackupFlow
  exitBackupFlow: typeof exitBackupFlow
}

type Props = StateProps & DispatchProps

const mapStateToProps = (state: RootState): StateProps => {
  return {
    language: state.app.language,
    backupCompleted: state.account.backupCompleted,
    backupTooLate: isBackupTooLate(state),
    backupDelayedTime: state.account.backupDelayedTime,
  }
}

export class Backup extends React.Component<Props, State> {
  static navigationOptions = { header: null }

  state = {
    mnemonic: '',
    currentQuestion: -1,
    wordsForBackupQuiz: [],
  }

  async componentDidMount() {
    this.props.enterBackupFlow()
    await this.retrieveMnemonic()
  }

  componentWillUnmount() {
    // Exit backup flow in unmount instead of cancel, as back button will not trigger onCancel().
    this.props.exitBackupFlow()
  }

  retrieveMnemonic = async () => {
    if (this.state.mnemonic) {
      return
    }

    try {
      const mnemonic = await getStoredMnemonic()
      if (!mnemonic) {
        throw new Error('Mnemonic not stored in key store')
      }
      this.setState({ mnemonic })

      const wordsForBackupQuiz = await createQuizWordList(mnemonic, this.props.language)
      this.setState({ wordsForBackupQuiz })
    } catch (e) {
      Logger.error('backup/retrieveMnemonic', e)
      // TODO(Rossy): use error banner
      Logger.showError('Error retrieving mnemonic')
    }
  }

  showBackupPhrase = () => {
    this.setState({ currentQuestion: 0 })
  }

  showQuiz = () => {
    this.setState({ currentQuestion: 1 })
  }

  showNextQuestion = () => {
    this.setState({ currentQuestion: this.state.currentQuestion + 1 })
  }

  returnToPhrase = () => {
    this.setState({ currentQuestion: 0 })
  }

  onCancel = async () => {
    navigateBack()
  }

  onDelay = () => {
    this.props.setBackupDelayed()
    navigateBack()
  }

  onFinish = async () => {
    this.props.setBackupCompleted()
    navigateBack()
  }

  render() {
    const { mnemonic, currentQuestion, wordsForBackupQuiz } = this.state
    const { backupCompleted, backupDelayedTime, backupTooLate } = this.props

    if (backupCompleted) {
      return (
        <BackupComplete
          backupCompleted={backupCompleted}
          onPress={this.onCancel}
          mnemonic={mnemonic}
        />
      )
    }

    if (currentQuestion === -1) {
      return (
        <BackupIntroduction
          onPress={this.showBackupPhrase}
          onCancel={this.onCancel}
          onDelay={this.onDelay}
          backupTooLate={backupTooLate}
          backupDelayedTime={backupDelayedTime}
        />
      )
    }
    if (currentQuestion === 0) {
      return (
        <BackupPhrase
          words={`${this.state.mnemonic}`}
          onPress={this.showQuiz}
          onCancel={this.onCancel}
        />
      )
    }
    if (mnemonic && currentQuestion > 0 && currentQuestion <= NUMBER_OF_TEST_QUESTIONS) {
      const indexToTest = INDICES_TO_TEST[currentQuestion - 1]
      const correctWord = mnemonic.split(' ')[indexToTest]
      const wordOptions = selectQuizWordOptions(
        correctWord,
        wordsForBackupQuiz,
        OPTIONS_PER_QUESTION
      )

      return (
        <BackupQuestion
          questionNumber={currentQuestion}
          testWordIndex={indexToTest}
          words={wordOptions}
          correctAnswer={correctWord}
          onReturnToPhrase={this.returnToPhrase}
          onCorrectSubmit={this.showNextQuestion}
          onWrongSubmit={this.returnToPhrase}
          onCancel={this.onCancel}
        />
      )
    }
    if (currentQuestion > NUMBER_OF_TEST_QUESTIONS) {
      return <BackupComplete onPress={this.onFinish} mnemonic={this.state.mnemonic} />
    }
  }
}

export default componentWithAnalytics(
  connect<StateProps, DispatchProps, {}, RootState>(
    mapStateToProps,
    { setBackupCompleted, setBackupDelayed, enterBackupFlow, exitBackupFlow }
  )(Backup)
)
