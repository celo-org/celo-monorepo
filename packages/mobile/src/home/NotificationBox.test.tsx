import * as React from 'react'
import { Provider } from 'react-redux'
import * as renderer from 'react-test-renderer'
import { DAYS_TO_BACKUP } from 'src/backup/Backup'
import NotificationBox from 'src/home/NotificationBox'
import { createMockStore } from 'test/utils'
import { mockPaymentRequests } from 'test/values'

const TWO_DAYS_MS = 2 * 24 * 60 * 1000

const storeData = {
  goldToken: { educationCompleted: true },
  account: {
    backupCompleted: true,
    dismissedEarnRewards: true,
    dismissedInviteFriends: true,
    accountCreationTime: new Date().getTime() - TWO_DAYS_MS,
    paymentRequests: [],
  },
}

describe('NotificationBox', () => {
  it('Simple test', () => {
    // const store = createMockStore(storeData)
    const store = createMockStore({
      ...storeData,
      account: {
        backupCompleted: false,
      },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <NotificationBox />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Backup too late test', () => {
    const store = createMockStore({
      ...storeData,
      account: {
        accountCreationTime: new Date().getTime() - DAYS_TO_BACKUP - TWO_DAYS_MS,
      },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <NotificationBox />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Educations not completed yet', () => {
    const store = createMockStore({
      ...storeData,
      goldToken: { educationCompleted: false },
      account: {
        backupCompleted: false,
        dismissedEarnRewards: false,
        dismissedInviteFriends: false,
        ...storeData.account,
      },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <NotificationBox />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Payment Requests exist', () => {
    const store = createMockStore({
      ...storeData,
      account: {
        ...storeData.account,
        paymentRequests: mockPaymentRequests,
      },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <NotificationBox />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
