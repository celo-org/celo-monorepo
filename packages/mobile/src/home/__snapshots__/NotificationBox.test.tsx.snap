// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBox Backup too late test 1`] = `
<View
  style={
    Object {
      "maxWidth": 750,
      "width": 750,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/backup-icon.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              backupKeyFlow6:yourBackupKey
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 5,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#81868B",
                        "fontFamily": "Hind-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      }
                    }
                  >
                    backupKeyFlow6:backupKeyNotification
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      backupKeyFlow6:getBackupKey
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/Invite-Friends.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              inviteFlow11:inviteFriendsToCelo
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 5,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#81868B",
                        "fontFamily": "Hind-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      }
                    }
                  >
                    inviteFlow11:inviteAnyone
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      global:inviteFriends
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      maybeLater
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#42D689",
          "borderRadius": 8,
          "flex": 0,
          "height": 8,
          "marginHorizontal": 5,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#D1D5D8",
          "borderRadius": 8,
          "flex": 0,
          "height": 6,
          "marginHorizontal": 5,
          "width": 6,
        }
      }
    />
  </View>
</View>
`;

exports[`NotificationBox Educations not completed yet 1`] = `
<View
  style={
    Object {
      "maxWidth": 750,
      "width": 750,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/home-icon.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              global:celoGold
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 5,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#81868B",
                        "fontFamily": "Hind-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      }
                    }
                  >
                    exchangeFlow9:whatIsGold
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      exchange
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      maybeLater
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NotificationBox Incoming Payment Requests exist 1`] = `
<View
  style={
    Object {
      "maxWidth": 750,
      "width": 750,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/send-dollar.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              incomingPaymentRequest
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 5,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      ellipsizeMode="middle"
                      numberOfLines={1}
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#81868B",
                              "fontFamily": "Hind-Regular",
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                          ]
                        }
                      >
                        +14155550000
                         - 
                        Rent request for June, it is already late!!!
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#81868B",
                              "fontFamily": "Hind-Regular",
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "fontFamily": "Hind-SemiBold",
                            },
                          ]
                        }
                      >
                        $12.34
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      review
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NotificationBox Outgoing Payment Requests exist 1`] = `
<View
  style={
    Object {
      "maxWidth": 750,
      "width": 750,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/send-dollar.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              outgoingPaymentRequest
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 5,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      ellipsizeMode="middle"
                      numberOfLines={1}
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#81868B",
                              "fontFamily": "Hind-Regular",
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                          ]
                        }
                      >
                        +14155550000
                         - 
                        Rent request for June, it is already late!!!
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#81868B",
                              "fontFamily": "Hind-Regular",
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "fontFamily": "Hind-SemiBold",
                            },
                          ]
                        }
                      >
                        $12.34
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      review
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NotificationBox Simple test 1`] = `
<View
  style={
    Object {
      "maxWidth": 750,
      "width": 750,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/backup-icon.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              backupKeyFlow6:yourBackupKey
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 5,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#81868B",
                        "fontFamily": "Hind-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      }
                    }
                  >
                    backupKeyFlow6:backupKeyNotification
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      backupKeyFlow6:getBackupKey
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 16,
            "width": 718,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 16,
                "width": "100%",
              },
              Object {
                "borderColor": "#EEEEEE",
                "borderRadius": 3,
                "borderWidth": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/mobile/src/images/Invite-Friends.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#2E3338",
                  "fontFamily": "Hind-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                }
              }
            >
              inviteFlow11:inviteFriendsToCelo
            </Text>
            <View>
              <View
                style={
                  Object {
                    "justifyContent": "space-between",
                    "minHeight": 70,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginTop": 5,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#81868B",
                        "fontFamily": "Hind-Regular",
                        "fontSize": 14,
                        "lineHeight": 20,
                      }
                    }
                  >
                    inviteFlow11:inviteAnyone
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 15,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      global:inviteFriends
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    nativeBackgroundAndroid={
                      Object {
                        "attribute": "selectableItemBackgroundBorderless",
                        "type": "ThemeAttrAndroid",
                      }
                    }
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#42D689",
                            "fontFamily": "Hind-SemiBold",
                            "fontSize": 14,
                            "lineHeight": 18,
                          },
                          Object {
                            "paddingEnd": 15,
                          },
                        ]
                      }
                    >
                      maybeLater
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#42D689",
          "borderRadius": 8,
          "flex": 0,
          "height": 8,
          "marginHorizontal": 5,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#D1D5D8",
          "borderRadius": 8,
          "flex": 0,
          "height": 6,
          "marginHorizontal": 5,
          "width": 6,
        }
      }
    />
  </View>
</View>
`;
